### ISSUE TITLE
Why `fatal` option is not supported?

### ISSUE THREAD

**Issue by khoanguyen-yang** [samthor/fast-text-encoding#24]
Hi,

I am wondering why the `fatal` option is not supported. Will setting `fatal=true` will cause any issue when using your library as a polyfill?

Many thanks

**Comment by samthor** [comment]
Setting `fatal=true` will cause the polyfill to fail since it's not supported.

I could probably add support, it wouldn't be too hard.

**Comment by khoanguyen-yang** [comment]
But it seems the `fatal` option is not used at all in the library

`fatal` is always set to `false` here https://github.com/samthor/fast-text-encoding/blob/master/src/o-decoder.js#L51
But the validation above it rejects the `fatal` option. https://github.com/samthor/fast-text-encoding/blob/master/src/o-decoder.js#L35
So wondering why you need the validation when you do not use it at all?

----

FYI, I am currently use your library to polyfill encoder/decoder in React Native and some other libraries do use the `fatal` option. I made a simple patch to disable only the validation (leaving `this.fatal=false` unchanged) and it seems to be working fine.
However I am not sure your library will work 100% ok for that change. So it would be nice if I have your confirmation or if you have time, you can make a change to the library if possible and necessary.

**Comment by pablodenadai** [comment]
> Setting fatal=true will cause the polyfill to fail since it's not supported.
> I could probably add support, it wouldn't be too hard.

That'd be awesome if you could add support for `fatal`. Otherwise please provide some guidance and I'll help implement it. Thanks!

**Comment by jupiterrosen** [comment]
@pablodenadai @samthor @khoanguyen-yang I'm also encountering this issue when integrating with solanaweb3js

**Comment by krisgrm** [comment]
yers please i need it also

**Comment by eduardhasanaj** [comment]
@samthor can you please let us know what is the status of this feature request?

**Comment by eduardhasanaj** [comment]
Till the pull request is approved you can use my branch
```
npm install github:eduardhasanaj/fast-text-encoding#add-fatal-option-support
```

**Comment by samthor** [comment]
The PR #29 doesn't cover all cases, i.e., it will not detect invalid UTF-8:
- it only covers one failure mode of UTF-8 encoding (it doesn't check continuation bytes) for the fallback code
- it doesn't support Node (`Buffer` does not throw, it generates U+FFFD characters for invalid input)

for folks on the thread, the two questions you need to ask are:
- do you actually want UTF-8 validation, _or do you just have some upstream library that expects `fatal` to be supported_
- why do you want this at all: modern Node and browsers do not need this polyfill (I wrote it in 2017!), and I'd go as far as saying including it is actively harmful.

**Comment by eduardhasanaj** [comment]
@samthor I agree that #29 does not cover all cases on detecting invalid utf-8 for the reasons you mentioned.

1. Some upstream library expects fatal mode to be supported.
2. I nedd this polyfill in React Native.

**Comment by deflexable** [comment]
i needed this polyfill so bson library can work on react-native

**Comment by ChALkeR** [comment]
@deflexable [@exodus/bytes/encoding.js](https://www.npmjs.com/package/@exodus/bytes#textencoder--textdecoder-polyfill) supports it and is ~10x faster than this package on React Native.
If you need just utf8, you can use `@exodus/bytes/utf8.js` which is tiny (and provides both fatal and non-fatal modes)


Also, native TextDecoder was just merged in Hermes and will be available at some point in React Native
